pkg/sfs/controllerserver.go

```diff
@@ -33,55 +37,88 @@ type controllerServer struct {
-	// Creating a share
-	createOpts := shares.CreateOpts{
-        ShareProto: cs.Driver.shareProto,
-        Size:       sizeInGiB,
-        Name:       req.GetName(),
-    }
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Failed to create SFS v2 client: %v", err)
+	}
+	name := req.GetName()
+	capacityRange := req.GetCapacityRange()
+	if err := createVolumeValidation(name, capacityRange); err != nil {
+		return nil, err
+	}
 
-    share, err := createShare(client, &createOpts)
+	requestedSize := capacityRange.GetRequiredBytes()
+	if requestedSize == 0 {
+		// At least 1GiB
+		requestedSize = 1 * common.GbByteSize
+	}
+	sizeInGiB := int(utils.RoundUpSize(requestedSize, common.GbByteSize))
+	share, err := checkVolumeExist(client, name, sizeInGiB, cs.Driver.shareProto)
 	if err != nil {
-		klog.V(3).Infof("Failed to create SFS volume: %v", err)
-		return nil, status.Error(codes.Internal, err.Error())
-    }
+		return nil, err
+	}
+	if share == nil {
+		// Creating a share
+		createOpts := shares.CreateOpts{
+			ShareProto: cs.Driver.shareProto,
+			Size:       sizeInGiB,
+			Name:       req.GetName(),
+		}
+		share, err = createShare(client, &createOpts)
+		if err != nil {
+			return nil, err
+		}
+	}
 
 	// Grant access to the share
-	klog.V(5).Infof("Creating an access ruleto VPC %s", cs.Driver.cloud.Vpc.Id)
-    if err := grantAccess(client, share.ID, cs.Driver.cloud.Vpc.Id); err != nil {
-		klog.V(3).Infof("Failed to create access rule for share: %v", err)
-		return nil, status.Error(codes.Internal, err.Error())
-    }
-
+	log.Infof("Creating an access rule to VPC %s", cs.Driver.cloud.Vpc.Id)
+	if err := grantAccess(client, share.ID, cs.Driver.cloud.Vpc.Id); err != nil {
+		return nil, err
+	}
 	return &csi.CreateVolumeResponse{
 		Volume: &csi.Volume{
 			VolumeId:      share.ID,
 			ContentSource: req.GetVolumeContentSource(),
-			CapacityBytes: int64(sizeInGiB) * bytesInGiB,
+			CapacityBytes: int64(sizeInGiB) * common.GbByteSize,
 		},
 	}, nil
 }
 
+func createVolumeValidation(name string, capacityRange *csi.CapacityRange) error {
+	if len(name) == 0 {
+		return status.Error(codes.InvalidArgument, "Validation failed, name cannot be empty")
+	}
+	if capacityRange == nil {
+		return status.Error(codes.InvalidArgument, "Validation failed, capacityRange cannot be nil")
+	}
+	return nil
+}
+
+func checkVolumeExist(client *golangsdk.ServiceClient, name string, sizeInGiB int, shareProto string) (
+	*shares.Share, error) {
+	opts := shares.ListOpts{
+		Name: name,
+	}
+	list, err := shareList(client, opts)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal,
+			"Failed to query SFS by name, cannot verify whether it exists: %v", err)
+	}
+	for _, v := range list {
+		if v.Name == name {
+			// if name, share proto and size are the same, return success
+			if v.ShareProto == shareProto && v.Size == sizeInGiB {
+				return &v, nil
+			}
+			return nil, status.Errorf(codes.InvalidArgument, "SFS name: %s already exists", name)
```

Comment:
suggest to print "SFS name: %s already exists with a different size or share_proto"

Reply:
done